plugins/inputs/mavlink/mavlink.go

```diff
@@ -0,0 +1,251 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package mavlink
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"net"
+	"net/url"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/bluenviron/gomavlib/v3"
+	"github.com/bluenviron/gomavlib/v3/pkg/dialects/ardupilotmega"
+	"github.com/bluenviron/gomavlib/v3/pkg/frame"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Mavlink struct {
+	URL                    string   `toml:"url"`
+	SystemID               uint8    `toml:"system_id"`
+	FilterPattern          []string `toml:"filter"`
+	StreamRequestDisable   bool     `toml:"stream_request_disable"`
+	StreamRequestFrequency uint16   `toml:"stream_request_frequency"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	filter         filter.Filter
+	connection     *gomavlib.Node
+	endpointConfig []gomavlib.EndpointConf
```

Comment:
Why do we need a slice here? Wouldn't a single endpoint config be sufficient?

Reply:
Honestly it's just because I was struggling with the Go syntax, I'll fix this now.

For what it's worth it is technically possible to use multiple endpoints. This is common in drone world where you may have a serial connection, TCP, and UDP connection all at once, networking together a system in multiple ways.

Reply:
Fix this so it's just one config instead of a slice.